app/controllers/charts_controller.rb

```diff
@@ -0,0 +1,22 @@
+class ChartsController < ApplicationController
+  include ApplicationHelper
```

Comment:
where is before_action? controller is 100% accessible

Reply:
Done.